en/docs/install-and-setup/upgrading-wso2-is-as-key-manager/upgrading-from-is-km-520-to-5100.md

```diff
@@ -10,60 +10,53 @@ The following information describes how to upgrade your **WSO2 API Manager (WSO2
         -   You are currently using a WSO2 IS 5.2.0 vanilla distribution that has WSO2 API Management related Key Manager features installed on top of it.
         -   You are currently using a pre-packaged WSO2 Identity Server 5.2.0 distribution.
 
-    -   If you wish to upgrade your APIM environment from **API-M 2.0.0 to 3.1.0**, which is using the **internal** WSO2 Identity Server (**WSO2 IS**) **capabilities**, follow the instructions in [Upgrading API-M from 2.0.0 to 3.1.0](../upgrading-wso2-api-manager/upgrading-from-200-to-310.md).
+    -   In prior to upgrade the **WSO2 Identity Server (WSO2 IS) AS Key Manager**, you have to upgrade your APIM environment from **API-M 2.0.0 to 3.1.0** as described in [Upgrading API-M from 2.0.0 to 3.1.0](../upgrading-wso2-api-manager/upgrading-from-200-to-310.md).
 
```

Comment:
We don't follow the whole steps in "Upgrading API-M from 2.0.0 to 3.1.0" before these steps right?

Reply:
This has been added if someone 1st come to this page and do the identity component migration first can be faced an issue during migration as they missed to execute the DB scripts related to AM table.

Reply:
What about this.  "Prior to upgrading the **WSO2 Identity Server (WSO2 IS) AS Key Manager**, you have to follow the required steps to upgrade your APIM environment from **API-M 2.0.0 to 3.1.0** as described in [Upgrading API-M from 2.0.0 to 3.1.0](../upgrading-wso2-api-manager/upgrading-from-200-to-310.md).

Reply:
"Prior to upgrading the WSO2 Identity Server (WSO2 IS) AS Key Manager, you have to follow the steps mentioned in "Upgrading API-M from 2.0.0 to 3.1.0." guide, to upgrade your APIM environment from API-M 2.0.0 to 3.1.0.